src/main/java/org/jabref/gui/groups/GroupTreeView.java

```diff
@@ -281,7 +299,13 @@ private void selectNode(GroupNodeViewModel value) {
         if (root.getValue().equals(value)) {
             return Optional.of(root);
         }
-        return root.getChildren().stream().filter(child -> getTreeItemByValue(child, value).isPresent()).findFirst();
+
+        Optional<TreeItem<GroupNodeViewModel>> node = Optional.empty();
+        for (int i = 0; i < root.getChildren().size() && node.isEmpty(); i++) {
```

Comment:
Can you please convert this to a normal foreach loop. I guess something like the following works:
```java
for (TreeItem<GroupNodeViewModel> child: root.getChildren()) {
    Optional<TreeItem<GroupNodeViewModel>> node = getTreeItemByValue(child, value);
    if (node.isPresent()) return node;
}
return Optional.empty();
```

Reply:
I don't see any difference to the stream variant? So why a loop?

Reply:
The stream returns the child of the root that contains the correct node, and not the node itself. The foreach can probably also be written as a stream...